GliaWidgets/SecureConversations/SecureConversations.ChatWithTranscriptModel.swift

```diff
@@ -402,32 +455,163 @@ extension SecureConversations {
+                        )
+                    case .noCameraPermission:
+                        self.showSettingsAlert(with: self.environment.alertConfiguration.cameraSettings)
+                    }
+                }
+                let file = ObservableValue<FilePickerEvent>(with: .none)
+                file.addObserver(self) { [weak self] event, _ in
+                    switch event {
+                    case .none, .cancelled:
+                        break
+                    case .pickedFile(let url):
+                        self?.filePicked(url)
+                    }
+                }
+                switch kind {
+                case .photoLibrary:
+                    self.delegate?(.pickMedia(media))
+                case .takePhoto:
+                    self.delegate?(.takeMedia(media))
+                case .browse:
+                    self.delegate?(.pickFile(file))
+                }
+            }
+            action?(.presentMediaPicker(itemSelected: { itemSelected($0) }))
+        }
+
+        private func mediaPicked(_ media: PickedMedia) {
+            switch media {
+            case .image(let url):
+                addUpload(with: url)
+            case .photo(let data, format: let format):
+                addUpload(with: data, format: format)
+            case .movie(let url):
+                addUpload(with: url)
+            case .none:
+                break
+            }
+        }
+
+        private func filePicked(_ url: URL) {
+            addUpload(with: url)
+        }
+
+        private func addUpload(with url: URL) {
+            guard let upload = fileUploadListModel.addUpload(with: url) else { return }
+            action?(.addUpload(upload))
+        }
+
+        private func addUpload(with data: Data, format: MediaFormat) {
+            guard let upload = fileUploadListModel.addUpload(with: data, format: format) else { return }
+            action?(.addUpload(upload))
+        }
+
+        private func removeUpload(_ upload: FileUpload) {
+            fileUploadListModel.removeUpload(upload)
+            action?(.removeUpload(upload))
+            validateMessage()
+        }
+
+        private func onUploaderStateChanged(_ state: FileUploader.State) {
+            validateMessage()
+        }
+
         @discardableResult
         private func validateMessage() -> Bool {
-            let canSendText = !messageText.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
-            // TODO: File uploading validation MOB-1742
-            action?(.sendButtonHidden(!canSendText))
-            return canSendText
+            let canSendText = !messageText.trimmingCharacters(in: .whitespacesAndNewlines)
+                .isEmpty && messageText.count <= Self.messageTextLimit
+            let canSendAttachments =
+            fileUploadListModel.failedUploads.isEmpty &&
+            fileUploadListModel.activeUploads.isEmpty && !fileUploadListModel.isLimitReached
```

Comment:
Looks like if maximum amount of files is attached, visitor can send the message. Am I right?

Reply:
No, visitor can not send if maximum is reached.